From 88429f56491ed0e5b3f5f91e6e352fa1251484bc Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 30 Nov 2020 08:55:34 +0000
Subject: [PATCH] v2v: windows: Fix schtasks /SD parameter.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Thanks: Dinesh Herath, Tomáš Golembiovský, Bryan Kinney,  Mark Zealey

Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1895323
(cherry picked from commit b65711c9293415f671d35d0e2f9b55a74343da45)
---
 v2v/convert_windows.ml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/v2v/convert_windows.ml b/v2v/convert_windows.ml
index 44cef5ed..4d844e2d 100644
--- a/v2v/convert_windows.ml
+++ b/v2v/convert_windows.ml
@@ -422,11 +422,12 @@ popd
  and configure_qemu_ga files =
    List.iter (
      fun msi_path ->
+       (* Windows is a trashfire. https://stackoverflow.com/a/18730884 *)
        let fb_script = sprintf "\
 echo Removing any previously scheduled qemu-ga installation
 schtasks.exe /Delete /TN Firstboot-qemu-ga /F
 echo Scheduling delayed installation of qemu-ga from %s
-powershell.exe -command \"$d = (get-date).AddSeconds(120); schtasks.exe /Create /SC ONCE /ST $d.ToString('HH:mm') /SD $d.ToString('MM/dd/yyyy') /RU SYSTEM /TN Firstboot-qemu-ga /TR \\\"C:\\%s /forcerestart /qn /l+*vx C:\\%s.log\\\"\"
+powershell.exe -command \"$d = (get-date).AddSeconds(120); $FormatHack = ($([System.Globalization.DateTimeFormatInfo]::CurrentInfo.ShortDatePattern) -replace 'M+/', 'MM/') -replace 'd+/', 'dd/'; schtasks.exe /Create /SC ONCE /ST $d.ToString('HH:mm') /SD $d.ToString($FormatHack) /RU SYSTEM /TN Firstboot-qemu-ga /TR \\\"C:\\%s /forcerestart /qn /l+*vx C:\\%s.log\\\"\"
       "
       msi_path msi_path msi_path in
       Firstboot.add_firstboot_script g inspect.i_root
